scripts/des_y3_sne_ia/des_y3_sne_ia.py

```diff
@@ -0,0 +1,190 @@
+            files = data_files
+            if isinstance(files, str):
+                files = [files]
+            # Use `dl_manager.iter_files` to skip hidden files in an extracted archive
+            files = [dl_manager.iter_files(file) for file in files]
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TRAIN, gen_kwargs={"files": files}
+                )
+            ]
+        splits = []
+        for split_name, files in data_files.items():
+            if isinstance(files, str):
+                files = [files]
+            # Use `dl_manager.iter_files` to skip hidden files in an extracted archive
+            files = [dl_manager.iter_files(file) for file in files]
+            splits.append(
+                datasets.SplitGenerator(name=split_name, gen_kwargs={"files": files})
+            )
+        return splits
+
+    def _generate_examples(self, files):
+        """Yields examples as (key, example) tuples."""
+        for file_number, file in enumerate(itertools.chain.from_iterable(files)):
+            with h5py.File(file, "r") as data:
+
+                keys = [data["object_id"][()]]
+
+                # Preparing an index for fast searching through the catalog
+                sort_index = np.argsort(data["object_id"][()])  # Accessing the scalar index
+                sorted_ids = [data["object_id"][()]]  # Ensure this is a list of one element
+
+                for k in keys:
+                    # Extract the indices of requested ids in the catalog
+                    i = sort_index[np.searchsorted(sorted_ids, k)]
+                    # Parse data
+                    idxs = np.arange(0, data["flux"].shape[0])
+                    band_idxs = idxs.repeat(data["flux"].shape[-1]).reshape(
+                         len(data["bands"][()].decode('utf-8').split(",")), -1
+                    )
+                    bands = data["bands"][()].decode('utf-8').split(",")
+                    example = {
+                        "lightcurve": {
+                            "band": np.asarray([bands[band_number] for band_number in band_idxs.flatten().astype("int32")]).astype("str"),
+                            "time": np.asarray(data["time"]).flatten().astype("float32"),
+                            "flux": np.asarray(data["flux"]).flatten().astype("float32"),
+                            "flux_err": np.asarray(data["flux_err"]).flatten().astype("float32"),
+                        }
+                    }
+                    # Add remaining features
+                    for f in _FLOAT_FEATURES:
+                        example[f] = np.asarray(data[f]).astype("float32")
+                    for f in _STR_FEATURES:
+                        # Add band names shared across dataset to each sample.
+                        # I can't see a better way to do this.
+                        if f == "bands":
+                            example[f] = data[f][()].decode('utf-8')
```

Comment:
why does this need `decode('utf-8')` again? it seems like when you make the `bands` array you already did `decoe('utf-8')`?